corporate/lib/stripe_event_handler.py

```diff
@@ -150,3 +152,34 @@ def handle_payment_intent_succeeded_event(
         False,
         billing_session.get_remote_server_legacy_plan(payment_intent.customer),
     )
+
+
+@error_handler
+def handle_invoice_paid_event(stripe_invoice: stripe.Invoice, invoice: Invoice) -> None:
+    invoice.status = Invoice.PAID
+    invoice.save(update_fields=["status"])
+
+    customer = invoice.customer
+
+    configured_fixed_price_plan = None
+    if customer.required_plan_tier:
+        configured_fixed_price_plan = get_configured_fixed_price_plan_offer(
+            customer, customer.required_plan_tier
+        )
+
+    if stripe_invoice.collection_method == "send_invoice" and configured_fixed_price_plan:
```

Comment:
Do we want/need to do logging when this function would exit without running this code? I wonder if we want to add some logging/assertions here.

Reply:
Actually, this function will be extended to handle other cases when this **if** condition is not met. That would be the part of the work Aman is doing (#28786) to [replace PaymentIntent with Invoice](https://chat.zulip.org/#narrow/stream/273-kandra-billing/topic/.240.20upgrade.20invoices/near/1729825) only.

So, I think, whatever logs/assertion we'd end up with will be finalized in #28786 ?

Reply:
OK that sounds reasonable to me. @amanagr FYI.